File: server/medicore.layer/MediCore.Service/Model/PatientModel.cs
using dna.core.service.Models;
using dna.core.service.Models.Abstract;
using MediCore.Data.Infrastructure;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Service.Model
{
    public class PatientModel : WriteHistoryBaseModel, IModelBase
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(100, ErrorMessage = "StringLength")]
        public string PatientName { get; set; }

        [Required(ErrorMessage = "Required")]        
        public RelationshipStatus RelationshipStatus { get; set; }

        [Required(ErrorMessage = "Required")]
        
        public PatientStatus PatientStatus { get; set; }

        [Required(ErrorMessage = "Required")]
        public Gender Gender { get; set; }

        public Nullable<DateTime> DateOfBirth { get; set; }


        //if patient is user
        public Nullable<int> AssociatedUserId { get; set; }

      
        public virtual UserModel AssociatedUser { get; set; }

    }
}

File: smartmedika.admin/src/app/model/patientModel.ts
import { IEntityBase } from './IEntityBase';
import { PatientStatus, RelationshipStatus, Gender } from './enumModel';
import { UserModel } from '../model';

export class PatientModel implements IEntityBase{
    id: number;
    patientName: string;
    relationshipStatus: RelationshipStatus;
    patientStatus: PatientStatus;
    gender: Gender;
    dateOfBirth?: string; 
    associatedUserId: number;
    associatedToUser: UserModel;    

    constructor(id: number, patientName: string, dateOfBirth?: string ) {
        this.id = 0;
        this.patientName = patientName;
        this.dateOfBirth = dateOfBirth;
        //this.patientStatus = null;
    }
}